app/services/clipboard/clipboard.ts

```diff
@@ -197,6 +197,18 @@ export class ClipboardService extends StatefulService<IClipboardState>
     return !!this.state.systemClipboard.files.length;
   }
 
+  canDuplicate(): boolean {
+    if (this.hasItemsInUnloadedClipboard()) return true;
+    if (!this.hasItems()) return false;
+    const hasNoduplicapableSource =
+      this.scenesService
+        .getScene(this.state.itemsSceneId)
+        .getSelection(this.state.sceneNodesIds)
+        .getSources()
+        .filter(source => source.doNotDuplicate).length > 0;
```

Comment:
not blocking but you can do `.some(source => source.doNotDuplicate)` here to make it a little easier to parse